GooglePlayInstant/Tests/Editor/QuickDeploy/OAuth2ServerTest.cs

```diff
@@ -0,0 +1,109 @@
+                "Expected valid code response");
+
+            // response is error
+            var errorResponse =
+                OAuth2Server.GetAuthorizationResponse(new Uri(string.Format("{0}?error={1}", AddressPrefix,
+                    someString)));
+            Assert.AreEqual(new KeyValuePair<string, string>("error", someString), errorResponse,
+                "Expected valid error response");
+
+            // response has invalid keys
+            var invalidResponse = new Uri(string.Format("{0}?someKey=someValue", AddressPrefix));
+            Assert.Throws<ArgumentException>(() => OAuth2Server.GetAuthorizationResponse(invalidResponse));
+
+            // No response. Uri has no params
+            Assert.Throws<ArgumentException>(() => OAuth2Server.GetAuthorizationResponse(new Uri(AddressPrefix)));
+        }
+
+        [Test]
+        public void TestGetQueryString()
+        {
+            var validUriWithCode = new Uri(string.Format("{0}?code=someValue", AddressPrefix));
+            Assert.AreEqual("code=someValue", OAuth2Server.GetQueryString(validUriWithCode));
+
+            var validUriWithError = new Uri(string.Format("{0}?error=someValue", AddressPrefix));
+            Assert.AreEqual("error=someValue", OAuth2Server.GetQueryString(validUriWithError));
+
+            var invalidUriWithEmptyQuery = new Uri(AddressPrefix);
+            Assert.Throws<ArgumentException>(() => OAuth2Server.GetQueryString(invalidUriWithEmptyQuery));
+        }
+
+        [Test]
+        public void TestGetCodeOrErrorResponsePairOnValidInputs()
+        {
+            Assert.AreEqual(new KeyValuePair<string, string>("code", "codeValue"),
+                OAuth2Server.GetCodeOrErrorResponsePair("code=codeValue"));
+
+            Assert.AreEqual(new KeyValuePair<string, string>("error", "errorValue"),
+                OAuth2Server.GetCodeOrErrorResponsePair("error=errorValue"));
+        }
+
+        [Test]
+        public void TestGetCodeOrErrorResponsePairOnInvalidInputs()
+        {
+            // The authorization response should consist of a single query parameter.
+            Assert.Throws<ArgumentException>(
+                () => OAuth2Server.GetCodeOrErrorResponsePair("code=codeValue&error=errorValue"));
+            Assert.Throws<ArgumentException>(
+                () => OAuth2Server.GetCodeOrErrorResponsePair("code=codeValue&otherKey=someValue"));
+            Assert.Throws<ArgumentException>(() =>
+                OAuth2Server.GetCodeOrErrorResponsePair("error=errorValue&otherKey=someValue"));
+
+            // The single query parameter should be an equals-separated key/value pair.
+            Assert.Throws<ArgumentException>(() => OAuth2Server.GetCodeOrErrorResponsePair("code&codeValue"));
+
+            // Key for the single query parameter should be either "code" or "error"
+            Assert.Throws<ArgumentException>(() => OAuth2Server.GetCodeOrErrorResponsePair("someKey=someValue"));
```

Comment:
Some potential test cases to add:
""
"&"
"="
"code"
"code=value1=value2"

Reply:
Resolved.